uaio assertion error in ws

After fixing #518 I've seen a new crash, which is an incorrect assertion in the code that assumes that the user supplied aio is not NULL.  If the the operation is canceled, this *can* be NULL, in which case we should treat the operation just like it failed.

```
panic: ../src/supplemental/websocket/websocket.c: 1259: assert err: uaio != NULL
This message is indicative of a BUG.
Report this at http://github.com/nanomsg/nanomsg
1   compat_ws                           0x000000010496aa6e nni_panic + 798
2   compat_ws                           0x00000001049c58c0 ws_http_cb_dialer + 512
3   compat_ws                           0x00000001049c3b16 ws_http_cb + 246
4   compat_ws                           0x0000000104978fac nni_taskq_thread + 348
5   compat_ws                           0x000000010497aa37 nni_thr_wrap + 503
6   compat_ws                           0x000000010498eeee nni_plat_thr_main + 606
7   libsystem_pthread.dylib             0x00007fff7eda1661 _pthread_body + 340
8   libsystem_pthread.dylib             0x00007fff7eda150d _pthread_body + 0
9   libsystem_pthread.dylib             0x00007fff7eda0bf9 thread_start + 13
Abort trap: 6
```

This was seen a couple of hours into a stress run.

--- a/src/supplemental/websocket/websocket.c
+++ b/src/supplemental/websocket/websocket.c
@@ -1256,11 +1256,13 @@ ws_http_cb_dialer(nni_ws *ws, nni_aio *aio)
 	uaio = ws->useraio;
 
 	nni_mtx_lock(&d->mtx);
-	NNI_ASSERT(uaio != NULL);
 	// We have two steps.  In step 1, we just sent the request,
 	// and need to retrieve the reply.  In step two we have
 	// received the reply, and need to validate it.
-	if ((rv = nni_aio_result(aio)) != 0) {
+	// Note that its possible that the user canceled the request,
+	// in which case we no longer care, and just go to the error
+	// case to discard the ws.
+	if (((rv = nni_aio_result(aio)) != 0) || (uaio == NULL)) {
 		goto err;
 	}
 
@@ -1343,7 +1345,9 @@ err:
 	if (nni_list_empty(&d->wspend)) {
 		nni_cv_wake(&d->cv);
 	}
-	nni_aio_finish_error(uaio, rv);
+	if (uaio != NULL) {
+		nni_aio_finish_error(uaio, rv);
+	}
 	nni_mtx_unlock(&d->mtx);
 
 	nni_ws_fini(ws);

